### PR TITLE
Base controller

### PR DESCRIPTION
**Pull request by denysxftr**

### CODE REVIEW

File: lib/zf/controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+private
+
+  attr_reader :request
+
+  def initialize(action_name)
+    @action = action_name.to_sym
+    @response_status = 200
+    @response_body = ''
+    @response_headers = {
+      'Date' => Time.now.httpdate
```
Comment (tilast) [inline lib/zf/controller.rb:29]: I'd add a global config like `timezone` and use it here

**Review by tilast — APPROVED** [review]

File: lib/zf/controller.rb (hunk 1 above)
Comment (denysxftr) [reply lib/zf/controller.rb:29]: no, it have to be in GMT https://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.3.1

File: lib/zf/controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    }
+  end
+
+  def response(param, content)
+    @response_headers.merge!('Content-Type' => RESPONSE_TYPES[param.to_sym][0])
+    @response_body = RESPONSE_TYPES[param.to_sym][1].call(content)
```
Comment (tilast) [inline lib/zf/controller.rb:35]: what if there is no such a response type?
it will raise the exception for `[]` method

Comment (denysxftr) [reply lib/zf/controller.rb:35]: yes for now

**Pull request merged**